rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java

```diff
@@ -129,61 +128,172 @@ void createInvalidLegacyErpFederation_emptyErpKeys() {
+
         ErpRedeemScriptBuilder builder = new NonStandardErpRedeemScriptBuilder();
         ErpFederationCreationException exception = assertThrows(
             ErpFederationCreationException.class,
-            () -> builder.createRedeemScriptFromKeys(standardKeys, defaultThreshold, emergencyKeys, emergencyThreshold, activationDelayValue)
+            () -> builder.createRedeemScriptFromKeys(
+                defaultKeys, defaultThreshold,
+                emergencyKeys, emergencyThreshold,
+                activationDelayValue)
         );
         assertEquals(INVALID_CSV_VALUE, exception.getReason());
     }
 
     @Test
     void createValidLegacyErpFederation_exactMaxCsvValue() {
         activationDelayValue = ErpRedeemScriptBuilderUtils.MAX_CSV_VALUE;
-        assertDoesNotThrow(this::createDefaultLegacyErpFederation);
+
+        ErpRedeemScriptBuilder builder = new NonStandardErpRedeemScriptBuilder();
+        assertDoesNotThrow(() -> builder
+            .createRedeemScriptFromKeys(
+                defaultKeys, defaultThreshold,
+                emergencyKeys, emergencyThreshold,
+                activationDelayValue));
     }
 
     @Test
-    void createInvalidNonStandardBuilder_aboveMaxScriptSigSize() {
+    void createErpRedeemScriptWithCsvUnsignedBE_csvValueOneByteLong_post_RSKIP284_pre_RSKIP293() {
+        // should create the redeem script with NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE
+        networkParameters = NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+
+        // For a value that only uses 1 byte it should add leading zeroes to complete 2 bytes
+        activationDelayValue = 20L;
+
+        createAndValidateFederation();
+        ErpRedeemScriptBuilder builder = federation.getErpRedeemScriptBuilder();
+        assertTrue(builder instanceof NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE);
+    }
+
+
+    @Test
+    void createValidNonStandardErpFederation_csvValueTwoBytesLong_post_RSKIP293() {
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        activationDelayValue = 500L;
+
+        createAndValidateFederation();
+        ErpRedeemScriptBuilder builder = federation.getErpRedeemScriptBuilder();
+        assertTrue(builder instanceof NonStandardErpRedeemScriptBuilder);
+    }
+
+    @Test
+    void createValidNonStandardErpFederation_csvValueTwoBytesLongIncludingSign_post_RSKIP293() {
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        activationDelayValue = 130; // Any value above 127 needs an extra byte to indicate the sign
+
+        createAndValidateFederation();
```

Comment:
Why are you calling this on most tests? Does it make any difference every time it is called? Can this be called once somewhere else?

Reply:
is different because the federation created is different each time